File: SkyRoute/src/SkyRoute.Application/Contracts/Requests/FlightSearchRequest.cs
using SkyRoute.Domain.Models;

namespace SkyRoute.Application.Contracts.Requests;

public sealed record FlightSearchRequest(
    string OriginAirportCode,
    string DestinationAirportCode,
    DateOnly DepartureDate,
    int Passengers,
    CabinClass CabinClass,
    string TimeZone);

File: SkyRoute.UI/src/app/models/flightSearchRequest.ts
export interface FlightSearchRequest {
  originAirportCode: string;
  destinationAirportCode: string;
  departureDate: string; // ISO format date string (e.g., "2024-12-31")
  passengers: number;
  cabinClass: 'Economy' | 'Business' | 'FirstClass';
  timeZone: string;
}